apps/omg_watcher_info/lib/omg_watcher_info/db/eth_event.ex

```diff
@@ -237,23 +240,26 @@ defmodule OMG.WatcherInfo.DB.EthEvent do
           event
       end
 
-    tx_output = resolve_tx_output(output_pointer)
+    case resolve_tx_output(output_pointer) do
+      %DB.TxOutput{} = tx_output ->
+        :ok = insert_exit_if_not_exist(ethevent, tx_output)
```

Comment:
why is this a case is it's supposed to exit?
wouldn't this be better?
```
tx_output = DB.TxOutput{}  = resolve_tx_output(output_pointer) 
:ok = insert_exit_if_not_exist(ethevent, tx_output)
```

Reply:
Unfortunately for one branch (IFE piggybacked outputs) we need to allow that outputs might not be present in the database. So the other `case` match checks for it.

Reply:
okay, so let's be very specific about that. let the code branch out if the exit involves `IFE piggybacked outputs`. this way the code will be more assertive about what is going on.

Reply:
https://github.com/omgnetwork/elixir-omg/pull/1769/commits/78f5ba4e99fd444b2d38a847c413a0afb3629db5